<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Peter Ralph">
  <title>Simulation-based inference for ecology and evolution</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="./reveal.js/dist/reset.css">
  <link rel="stylesheet" href="./reveal.js/dist/reveal.css">
  <style>
    .reveal .sourceCode {  /* see #7635 */
      overflow: visible;
    }
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <link rel="stylesheet" href="./reveal.js/dist/theme/simple.css" id="theme">
  <style type="text/css">

  .reveal { font-size: 30px; }

  .reveal h1 { font-size: 1.5em; } 

  .reveal h2 { font-size: 1.2em; } 

  .reveal .slides { text-align: left; }

  .reveal .slides figure { text-align: center; }

  .centered p { text-align: center; }

  /* apply this to the columns div */
  div.vcentered { display: flex; align-items: center; }

  div.bottomright { position: fixed; bottom: 0; right: 0; }

  .caption p { font-size: 75%; font-style: italic; color: #0006; clear: both; }

  .greyed { color: #4444; }

  img.floatright, div.floatright { float: right; text-align: right; }
  img.floatleft, div.floatleft { float: left; text-align: left; }

  img.centered { margin-left: auto; margin-right: auto; display: block; }

  .invisible { opacity: 0; }

  .reveal figcaption { display: none; }

  .reveal img { border: none; }

  .reveal section img { border: none; }

  .MathJax_Display { margin: 0; }

  img.finger { vertical-align: text-top; }

  .smaller { font-size: 50% }
  .smallish { font-size: 80% }

  </style>

  <div style="display: none">
  \[
  \newcommand{\E}{\mathbb{E}}
  \renewcommand{\P}{\mathbb{P}}
  \DeclareMathOperator{\var}{var}
  \]
  </div>
</head>
<body>
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">Simulation-based inference for ecology and
evolution</h1>
  <p class="author">Peter Ralph</p>
  <p class="date">Department of Statistics <br/> Oregon State University
// 26 Feb 2024</p>
</section>

<section class="slide level2">

<blockquote>
<p>Both the UO and OSU are located on the traditional indigenous
homeland of the Kalapuya people. Kalapuya people were dispossessed of
their indigenous homeland by the United States government and forcibly
removed. Today, Kalapuya descendants are primarily citizens of the
Confederated Tribes of Grand Ronde and the Confederated Tribes of Siletz
Indians, and continue to make important contributions to their
communities, to the UO, to Oregon, and to the world.</p>
</blockquote>
</section>
<section>
<section id="outline" class="title-slide slide level1">
<h1>Outline</h1>

</section>
<section id="outline-of-the-talk" class="slide level2">
<h2>Outline of the talk</h2>
<ol type="1">
<li>Genomes, and simulations</li>
<li>The tree sequence</li>
<li>Applications</li>
</ol>
<p><em>slides:</em> <a
href="https://petrelharp.github.io/corvallis-feb-2024/corvallis-feb-2024.slides.html">github.com/petrelharp/corvallis-feb-2024</a></p>
</section>
<section id="motivating-problems" class="slide level2">
<h2>Motivating problems</h2>
<div class="columns">
<div class="column" style="width:80%;">
<ol type="1">
<li><p>How much natural selection is needed to explain the correlations
in genetic diversity along the genomes of the great apes?</p></li>
<li><p>How can we predict the landscape of population density using
spatial genomic data?</p></li>
</ol>
</div><div class="column" style="width:20%;">
<p><img data-src="figs/modeling_empty.png" /></p>
</div>
</div>
<p><img data-src="figs/murillo/greatapes_pi.png" /></p>
</section></section>
<section>
<section id="inference-with-genomes" class="title-slide slide level1">
<h1>Inference, with genomes</h1>

</section>
<section id="section" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_empty.png" /></p>
</section>
<section id="section-1" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_reality.png" /></p>
</section>
<section id="section-2" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_model_parameters.png" /></p>
</section>
<section id="section-3" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_model_parameters_inverse.png" /></p>
</section>
<section id="section-4" class="slide level2">
<h2></h2>
<p><img data-src="figs/modeling_parameters_inverse_computer.png" /></p>
</section>
<section id="simulation-based-inference" class="slide level2">
<h2>Simulation-based inference</h2>
<div class="centered">
<p><img data-src="figs/modeling_parameters_inverse_computer.png"
style="width:60.0%" /></p>
</div>
<ul>
<li>bespoke confirmatory simulations</li>
<li>optimization of one or two parameters
<!-- *(if between-simulation noise is small)* --></li>
<li>Approximate Bayesian Computation (ABC)</li>
<li>deep learning</li>
</ul>
</section></section>
<section>
<section id="what-do-we-need" class="title-slide slide level1">
<h1>What do we need</h1>

</section>
<section id="section-5" class="slide level2">
<h2></h2>
<ol type="1">
<li><p>Fast simulation of genomes</p></li>
<li><p>Fast computation of summary statistics</p></li>
</ol>
</section>
<section id="wish-list" class="slide level2">
<h2>Wish list:</h2>
<div class="smallish">
<div class="columns">
<div class="column">
<p>Whole genomes, thousands of samples,<br />
from millions of individuals.</p>
<p><strong>Demography:</strong></p>
<ul>
<li>life history</li>
<li>separate sexes</li>
<li>selfing</li>
<li>polyploidy</li>
<li>species interactions</li>
</ul>
<p><strong>Geography:</strong></p>
<ul>
<li>discrete populations</li>
<li>continuous landscapes</li>
<li>barriers</li>
</ul>
<p><strong>History:</strong></p>
<ul>
<li>ancient samples</li>
<li>range shifts</li>
</ul>
</div><div class="column">
<p><strong>Natural selection:</strong></p>
<ul>
<li>selective sweeps</li>
<li>introgressing alleles</li>
<li>background selection</li>
<li>quantitative traits</li>
<li>incompatibilities</li>
<li>local adaptation</li>
</ul>
<p><strong>Genomes:</strong></p>
<ul>
<li>recombination rate variation</li>
<li>gene conversion</li>
<li>infinite-sites mutation</li>
<li>nucleotide models</li>
<li>context-dependence</li>
<li>mobile elements</li>
<li>inversions</li>
<li>copy number variation</li>
</ul>
</div>
</div>
</div>
</section>
<section id="enter-slim" class="slide level2">
<h2>Enter SLiM</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p>by Ben Haller and Philipp Messer</p>
<p>an individual-based, scriptable forwards simulator</p>
<div class="caption">
<p><img data-src="figs/ben-haller.jpg" alt="Ben Haller" /> <em>Ben
Haller</em></p>
</div>
</div><div class="column" style="width:50%;">
<figure>
<img data-src="figs/slim-gui.png" alt="SLiM GUI" />
<figcaption aria-hidden="true">SLiM GUI</figcaption>
</figure>
<p><a href="https://messerlab.org/SLiM/">messerlab.org/SLiM</a></p>
</div>
</div>
</section>
<section id="section-6" class="slide level2">
<h2></h2>
<div class="smallish">
<div class="columns">
<div class="column">
<ul>
<li><s>Whole genomes,</s>*</li>
<li><s>thousands of samples, </s></li>
<li><s>from millions of individuals.</s>*</li>
</ul>
<p><strong>Demography:</strong></p>
<ul>
<li><s>life history</s></li>
<li><s>separate sexes</s>*</li>
<li><s>selfing</s></li>
<li>polyploidy*</li>
<li>species interactions <strong>(coming soon!)</strong></li>
</ul>
<p><strong>Geography:</strong></p>
<ul>
<li><s>discrete populations</s></li>
<li><s>continuous landscapes</s></li>
<li><s>barriers</s>*</li>
</ul>
<p><strong>History:</strong></p>
<ul>
<li><s>ancient samples</s></li>
<li><s>range shifts</s></li>
</ul>
</div><div class="column">
<p><strong>Natural selection:</strong></p>
<ul>
<li><s>selective sweeps</s></li>
<li><s>introgressing alleles</s></li>
<li><s>background selection</s></li>
<li><s>quantitative traits</s>*</li>
<li><s>incompatibilities</s>*</li>
<li><s>local adaptation</s>*</li>
</ul>
<p><strong>Genomes:</strong></p>
<ul>
<li><s>recombination rate variation</s></li>
<li><s>gene conversion</s></li>
<li><s>infinite-sites mutation</s></li>
<li><s>nucleotide models</s></li>
<li><s>context-dependence</s>*</li>
<li>mobile elements*</li>
<li>inversions*</li>
<li>copy number variation</li>
</ul>
</div>
</div>
</div>
</section>
<section id="section-7" class="slide level2">
<h2></h2>
<ul>
<li><s>Whole genomes,</s>*</li>
</ul>
</section></section>
<section>
<section id="genomes-and-genealogies" class="title-slide slide level1">
<h1>Genomes and genealogies</h1>

</section>
<section id="genomes" class="slide level2">
<h2>Genomes</h2>
<ul>
<li class="fragment">are very big (<span
class="math inline">\(10^7\)</span>–<span
class="math inline">\(10^{12}\)</span> nucleotides)</li>
<li class="fragment">encode the basic <em>mechanisms</em> of life</li>
<li class="fragment">reflect past <em>history</em> and
<em>process</em></li>
</ul>
<div class="fragment">
<figure>
<img data-src="figs/kreitman-1983.png"
alt="Genotype matrix from Kreitman 1983" />
<figcaption aria-hidden="true">Genotype matrix from Kreitman
1983</figcaption>
</figure>
</div>
</section>
<section id="section-8" class="slide level2">
<h2></h2>
<p>Your two copies of the genome were inherited, noisily, from your two
parents,</p>
<div class="fragment">
<p>and from your four grandparents,</p>
</div>
<div class="fragment">
<p>and your eight great-grandparents</p>
</div>
<div class="fragment">
<p>and your sixteen great-great-grandparents</p>
<!--

## Genealogical ancestors

In a well-mixed population of size $N$,

- A most recent common ancestor of *everyone* lived at $\approx \log_2 N$ generations ago.

- For humans: $\approx 500$ years; 
    but including geography and history,
    $\approx$ 2--3 thousand years
    (Chang ’99; Rohde, Olson & Chang ’04)

- *Everyone* alive $\approx 1.77 \log_2 N$ generations ago
    is either an ancestor of everyone or has left behind no descendants. (Chang ’99)

-->
</div>
</section>
<section id="meiosis-recombination" class="slide level2">
<h2>Meiosis &amp; Recombination</h2>
<div class="columns">
<div class="column" style="width:50%;">
<figure>
<img data-src="figs/recombination-cartoon-crop.png"
alt="recombination" />
<figcaption aria-hidden="true">recombination</figcaption>
</figure>
</div><div class="column" style="width:50%;">
<ul>
<li><p>You have two copies of each chromosome, one from each
parent.</p></li>
<li><p>When you make a gamete, the copies <em>recombine</em>, at
Poisson(<span class="math inline">\(\rho\)</span>) locations.</p></li>
<li><p><em>Mutations</em> appear at Poisson(<span
class="math inline">\(\mu\)</span>) locations.</p></li>
</ul>
</div>
</div>
</section>
<section id="you-and-your-greatk-2-grand-parents" class="slide level2">
<h2>You and your great<span
class="math inline">\({}^{k-2}\)</span>-grand parents</h2>
<p>The number of <span class="math inline">\(k^\text{th}\)</span>
<em>genetic</em> ancestors grows <em>linearly</em> with <span
class="math inline">\(k\)</span>.</p>
<div class="centered">
<p><img
data-src="figs/gcbias/num_genetics_vs_genealogical_ancs.png" /></p>
</div>
<div class="caption">
<p>from <a
href="https://gcbias.org/2013/11/11/how-does-your-number-of-genetic-ancestors-grow-back-over-time/">gcbias.org</a></p>
</div>
</section>
<section id="section-9" class="slide level2">
<h2></h2>
<div class="centered">
<p><img data-src="figs/gcbias/parents_grandparents.png"
style="width:60.0%" /></p>
</div>
<div class="caption">
<p>from <a
href="https://gcbias.org/2013/11/11/how-does-your-number-of-genetic-ancestors-grow-back-over-time/">gcbias.org</a></p>
</div>
</section>
<section id="section-10" class="slide level2">
<h2></h2>
<div class="centered">
<p><img data-src="figs/gcbias/male_line.png" style="width:60.0%" /></p>
</div>
<div class="caption">
<p>from <a
href="https://gcbias.org/2013/11/11/how-does-your-number-of-genetic-ancestors-grow-back-over-time/">gcbias.org</a></p>
</div>
<!-- Tree sequences -->
</section></section>
<section>
<section id="the-tree-sequence" class="title-slide slide level1">
<h1>The tree sequence</h1>

</section>
<section id="history-is-a-sequence-of-trees" class="slide level2">
<h2>History is a sequence of trees</h2>
<p>For a set of sampled chromosomes, at each position along the genome
there is a genealogical tree that says how they are related.</p>
<div class="fragment">
<figure>
<img data-src="figs/example.svg" alt="Trees along a chromosome" />
<figcaption aria-hidden="true">Trees along a chromosome</figcaption>
</figure>
</div>
</section>
<section class="slide level2">

<p>The <strong>succinct tree sequence</strong></p>
<div class="floatright">
<p>is a way to succinctly describe this, er, sequence of trees</p>
<p><em>and</em> the resulting genome sequences.</p>
<div class="caption">
<p><a
href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004842">Kelleher,
Etheridge, &amp; McVean</a></p>
</div>
</div>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<figure>
<img data-src="figs/tskit_logo.png" style="width:80.0%"
alt="tskit logo" />
<figcaption aria-hidden="true">tskit logo</figcaption>
</figure>
</div><div class="column" style="width:50%;">
<div class="floatright">
<figure>
<img data-src="figs/jerome.jpeg" style="width:50.0%"
alt="jerome kelleher" />
<figcaption aria-hidden="true">jerome kelleher</figcaption>
</figure>
<div class="caption">
<p>jerome kelleher</p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="example-three-samples-two-trees-two-variant-sites"
class="slide level2">
<h2>Example: three samples; two trees; two variant sites</h2>
<figure>
<img data-src="figs/example_tree_sequence.png"
alt="Example tree sequence" />
<figcaption aria-hidden="true">Example tree sequence</figcaption>
</figure>
</section>
<section id="nodes-and-edges" class="slide level2">
<h2>Nodes and edges</h2>
<dl>
<dt>Edges</dt>
<dd>
<p>Who inherits from who.</p>
<p>Records: interval (left, right); parent node; child node.</p>
</dd>
<dt>Nodes</dt>
<dd>
<p>The ancestors those happen in.</p>
<p>Records: time ago (of birth); ID (implicit).</p>
</dd>
</dl>
</section>
<section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.0.png"
alt="Building a tree sequence" />
<figcaption aria-hidden="true">Building a tree sequence</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.1.png"
alt="Building a tree sequence" />
<figcaption aria-hidden="true">Building a tree sequence</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.2.png"
alt="Building a tree sequence" />
<figcaption aria-hidden="true">Building a tree sequence</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.3.png"
alt="Building a tree sequence" />
<figcaption aria-hidden="true">Building a tree sequence</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.4.png"
alt="Building a tree sequence" />
<figcaption aria-hidden="true">Building a tree sequence</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.5.png"
alt="Building a tree sequence" />
<figcaption aria-hidden="true">Building a tree sequence</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="nodes_edges_walkthrough/nodes_edges_walkthrough.6.png"
alt="Building a tree sequence" />
<figcaption aria-hidden="true">Building a tree sequence</figcaption>
</figure>
</section>
<section id="sites-and-mutations" class="slide level2">
<h2>Sites and mutations</h2>
<dl>
<dt>Mutations</dt>
<dd>
<p>When state changes along the tree.</p>
<p>Records: site it occured at; node it occurred in; derived state.</p>
</dd>
<dt>Sites</dt>
<dd>
<p>Where mutations fall on the genome.</p>
<p>Records: genomic position; ancestral (root) state; ID (implicit).</p>
</dd>
</dl>
</section>
<section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.0.png"
alt="Adding mutations" />
<figcaption aria-hidden="true">Adding mutations</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.1.png"
alt="Adding mutations" />
<figcaption aria-hidden="true">Adding mutations</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.2.png"
alt="Adding mutations" />
<figcaption aria-hidden="true">Adding mutations</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.3.png"
alt="Adding mutations" />
<figcaption aria-hidden="true">Adding mutations</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="sites_muts_walkthrough/sites_muts_walkthrough.4.png"
alt="Adding mutations" />
<figcaption aria-hidden="true">Adding mutations</figcaption>
</figure>
</section>
<section class="slide level2">

<p><strong>The result:</strong> an encoding of the genomes <em>and</em>
all the genealogical trees.</p>
<div class="centered">
<figure>
<img data-src="figs/example_tree_sequence.png"
alt="Example tree sequence" />
<figcaption aria-hidden="true">Example tree sequence</figcaption>
</figure>
</div>
</section></section>
<section>
<section id="hows-it-work" class="title-slide slide level1">
<h1>How’s it work?</h1>

</section>
<section id="file-sizes" class="slide level2">
<h2>File sizes</h2>
<div class="centered">
<figure>
<img data-src="figs/tsinfer_sizes.png" style="width:90.0%"
alt="file sizes" />
<figcaption aria-hidden="true">file sizes</figcaption>
</figure>
</div>
<div class="caption">
<p>100Mb chromosomes; from <a
href="https://www.nature.com/articles/s41588-019-0483-y">Kelleher et al
2018, <em>Inferring whole-genome histories in large population
datasets</em></a>, Nature Genetics</p>
</div>
<!-- Estimated sizes of files required to store the genetic variation data for a
simulated human-like chromosome (100 megabases) for up to 10 billion haploid
(5 billion diploid) samples. Simulations were run for 10 1 up to 10 7 haplotypes
using msprime [Kelleher et al., 2016], and the sizes of the resulting files plotted
(points). -->
</section>
<section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_genotypes.png"
alt="genotypes" />
<figcaption aria-hidden="true">genotypes</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_genotype_and_tree.png"
alt="genotypes and a tree" />
<figcaption aria-hidden="true">genotypes and a tree</figcaption>
</figure>
</section>
<section class="slide level2">

<figure>
<img data-src="figs/ts_ex/tree_sequence_next_genotype_and_tree.png"
alt="genotypes and the next tree" />
<figcaption aria-hidden="true">genotypes and the next tree</figcaption>
</figure>
</section>
<section id="for-n-samples-genotyped-at-m-sites" class="slide level2">
<h2>For <span class="math inline">\(N\)</span> samples genotyped at
<span class="math inline">\(M\)</span> sites</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><em>Genotype matrix</em>:</p>
<p><span class="math inline">\(N \times M\)</span> things.</p>
</div><div class="column" style="width:50%;">
<p><em>Tree sequence:</em></p>
<ul>
<li><span class="math inline">\(2N-2\)</span> edges for the first
tree</li>
<li><span class="math inline">\(\sim 4\)</span> edges per each of <span
class="math inline">\(T\)</span> trees</li>
<li><span class="math inline">\(M\)</span> mutations</li>
</ul>
<p><span class="math inline">\(O(N + T + M)\)</span> things</p>
</div>
</div>
<figure>
<img data-src="figs/ts_ex/tree_sequence_genotype_and_tree.png"
style="width:60.0%" alt="genotypes and a tree" />
<figcaption aria-hidden="true">genotypes and a tree</figcaption>
</figure>
</section></section>
<section>
<section id="summarizing-genomes-and-genealogies"
class="title-slide slide level1">
<h1>Summarizing genomes and genealogies</h1>

</section>
<section id="fast-genotype-statistics" class="slide level2">
<h2>Fast genotype statistics</h2>
<div class="centered">
<figure>
<img data-src="figs/treestats/benchmarks_without_copy_longer_genome.png"
style="width:70.0%" alt="efficiency of treestat computation" />
<figcaption aria-hidden="true">efficiency of treestat
computation</figcaption>
</figure>
</div>
<div class="caption">
<p>from <a
href="https://academic.oup.com/genetics/article/215/3/779/5930459">R.,
Thornton and Kelleher 2019, <em>Efficiently summarizing relationships in
large samples</em></a>, Genetics</p>
</div>
</section>
<section id="summaries-of-genotypes-and-genealogies"
class="slide level2">
<h2>Summaries of genotypes and genealogies</h2>
<div class="columns">
<div class="column" style="width:47%;">
<p><em>Genotypes:</em></p>
<ol type="1">
<li>For each site,</li>
<li>look at who has which alleles,</li>
<li>and add a <em>summary</em> of these values to our running
total.</li>
</ol>
<p><em>Example:</em> genetic distance counts how many mutations differ
between two sequences.</p>
</div><div class="column" style="width:5%;">

</div><div class="column" style="width:47%;">
<figure>
<img data-src="figs/ts_ex/tree_sequence_site.png" alt="site stats" />
<figcaption aria-hidden="true">site stats</figcaption>
</figure>
</div>
</div>
</section>
<section id="summaries-of-genotypes-and-genealogies-1"
class="slide level2">
<h2>Summaries of genotypes and genealogies</h2>
<div class="columns">
<div class="column" style="width:47%;">
<p><em>Trees:</em></p>
<ol type="1">
<li>For each branch,</li>
<li>look at who would inherit mutations on that branch,</li>
<li>and add the <em>expected contribution</em> to the running
total.</li>
</ol>
<p><em>Example:</em> the mean time to most recent common ancestor
between two sequences.</p>
</div><div class="column" style="width:5%;">

</div><div class="column" style="width:47%;">
<figure>
<img data-src="figs/ts_ex/tree_sequence_branch.png"
alt="branch stats" />
<figcaption aria-hidden="true">branch stats</figcaption>
</figure>
</div>
</div>
</section>
<section id="section-11" class="slide level2">
<h2></h2>
<p>Given</p>
<ol type="1">
<li>a <em>weight</em> <span class="math inline">\(w_i \in
\mathbb{R}^n\)</span> for each <em>sample node</em>, and</li>
<li>a <em>summary function</em> <span class="math inline">\(f :
\mathbb{R}^n \to \mathbb{R}\)</span>,</li>
</ol>
<div class="fragment">
<p>the <strong>Site</strong> statistic <span
class="math display">\[\begin{equation}
\text{Site}(f,w) = \sum_{i: \text{sites}} \sum_{a: \text{alleles}_i}
f(w_a)
\end{equation}\]</span> is the total summarized weights of all
mutations,</p>
</div>
<div class="fragment">
<p>and the <strong>Branch</strong> statistic <span
class="math display">\[\begin{equation}
\text{Branch}(f,w) = \sum_{T: \text{trees}} s_T \sum_{b:
\text{branches}_T} f(w_b) \ell_b
\end{equation}\]</span> is the <em>expected value</em> of <span
class="math inline">\(\text{Site}(f,w)\)</span> under Poisson(1)
mutation, given the trees.</p>
</div>
</section>
<section id="section-12" class="slide level2">
<h2></h2>
<p>With genealogies <em>fixed</em>, and averaging only over
<em>mutations</em> with rate <span class="math inline">\(\mu\)</span>,
<span class="math display">\[\begin{equation}
    \text{Branch}(f, w) = \frac{1}{\mu} \mathbb{E}\left[ \text{Site}(f,
w) \vert T \right] .
\end{equation}\]</span></p>
<div class="fragment">
<p>Dealing directly with genealogies removes the layer of noise due to
mutation: <span class="math display">\[\begin{equation}
    \frac{1}{\mu^2} \var\left[\text{Site}(f,w)\right]
    =
    \var\left[\text{Branch}(f,w)\right]
    +
    \frac{1}{n}
    \mathbb{E}\left[\text{Branch}(f^2,w)\right]
\end{equation}\]</span></p>
</div>
<div class="fragment">
<p>and might produce <em>unbiased</em> estimates from ascertained
genotype data.</p>
<div class="caption greyed floatright">
<p>also see R., TPB, 2019</p>
</div>
</div>
</section>
<section id="section-13" class="slide level2">
<h2></h2>
<div class="columns">
<div class="column" style="width:60%;">
<figure>
<img
data-src="figs/treestats/relate_chr20_site_div_branch.1000000.diversity.png"
style="width:100.0%" alt="duality in 1000G data" />
<figcaption aria-hidden="true">duality in 1000G data</figcaption>
</figure>
</div><div class="column" style="width:40%;">
<p>Duality, on 1000 Genomes data? Not quite…</p>
<ul>
<li>variation in mutation rate?</li>
<li>biased gene conversion?</li>
<li>selection?</li>
<li>inference artifacts?</li>
</ul>
<div class="caption">
<p><em>Tree sequence from <a
href="https://www.nature.com/articles/s41588-019-0484-x">Speidel et al
2019</a>.</em></p>
</div>
</div>
</div>
</section></section>
<section>
<section id="application-to-genomic-simulations"
class="title-slide slide level1">
<h1>Application to genomic simulations</h1>

</section>
<section id="the-main-idea" class="slide level2">
<h2>The main idea</h2>
<p>If we <em>record the tree sequence</em> that relates everyone to
everyone else,</p>
<p>after the simulation is over we can put neutral mutations down on the
trees.</p>
<div class="fragment">
<p>Since neutral mutations don’t affect demography,</p>
<p>this is <em>equivalent</em> to having kept track of them
throughout.</p>
</div>
</section>
<section class="slide level2">

<p>This means recording the entire genetic history of
<strong>everyone</strong> in the population, <strong>ever</strong>.</p>
<div class="fragment">
<p>It is <em>not</em> clear this is a good idea.</p>
</div>
<div class="fragment">
<p>But, with a few tricks…</p>
</div>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<div class="caption">
<p>From Kelleher, Thornton, Ashander, and R. 2018, <a
href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006581">Efficient
pedigree recording for fast population genetics simulation</a>.</p>
<p>and Haller, Galloway, Kelleher, Messer, and R. 2018, <a
href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1755-0998.12968"><em>Tree‐sequence
recording in SLiM opens new horizons for forward‐time simulation of
whole genomes</em></a></p>
</div>
</div><div class="column" style="width:50%;">
<div class="floatright">
<p><img data-src="figs/jared.jpeg" style="width:35.0%"
alt="jared galloway" /> <img data-src="figs/jaime.jpg"
style="width:30.0%" alt="jaime ashander" /> <img
data-src="figs/ben-haller.jpg" style="width:30.0%"
alt="ben haller" /></p>
</div>
</div>
</div>
</div>
</section>
<section id="a-100x-speedup" class="slide level2">
<h2>A 100x speedup!</h2>
<div class="columns">
<div class="column" style="width:40%;">
<div class="centered">
<figure>
<img data-src="figs/slim_logo.png" style="width:100.0%"
alt="SLiM logo" />
<figcaption aria-hidden="true">SLiM logo</figcaption>
</figure>
</div>
</div><div class="column" style="width:60%;">
<div class="floatright">
<p><img data-src="figs/slim_timing_no_msprime.png"
style="width:100.0%" /></p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<p><strong>For example:</strong></p>
<ul>
<li>genome as human chr7 (<span class="math inline">\(1.54 \times
10^8\)</span>bp)</li>
<li><span class="math inline">\(\approx\)</span> 10,000 diploids</li>
<li>500,000 overlapping generations</li>
<li>continuous, square habitat</li>
<li>selected mutations at rate <span
class="math inline">\(10^{-10}\)</span></li>
<li>neutral mutations added afterwards</li>
</ul>
<p><em>Runtime:</em> 8 hours on a single core</p>
<!-- Great apes -->
</section></section>
<section>
<section id="landscapes-of-genetic-diversity"
class="title-slide slide level1">
<h1>Landscapes of genetic diversity</h1>
<p><a
href="https://academic.oup.com/genetics/advance-article/doi/10.1093/genetics/iyae006/7577595"><img
data-src="figs/murillo/genetics.png" class="floatleft"
style="width:70.0%"
alt="https://academic.oup.com/genetics/advance-article/doi/10.1093/genetics/iyae006/7577595" /></a></p>
<div class="caption floatright">
<p><img data-src="figs/murillo.jpeg" style="width:80.0%" /></p>
<p><a href="https://www.m-rodrigues.me/">Murillo Rodrigues</a></p>
</div>
</section>
<section id="landscapes-of-genetic-diversity-within-groups-of-species"
class="slide level2">
<h2>Landscapes of genetic diversity within groups of species</h2>
<p><img data-src="figs/murillo/correlated-pi-examples.png"
style="width:80.0%" /></p>
<div class="caption floatright">
<p><a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000391">Stankowski,
Chase et al. (2019)</a>; <a
href="https://onlinelibrary.wiley.com/doi/10.1111/mec.14083">Van Doren
et al. (2017)</a>; <a
href="https://academic.oup.com/evlett/article/1/3/118/6697077">Buri
(2017)</a></p>
</div>
</section>
<section id="the-study-system-us" class="slide level2">
<h2>The study system: “us”</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>High quality genomic data for 5 species</li>
<li>Deep divergence, spanning ~60N generations</li>
<li>Conservation of genes, recombination rate and other genomic
features</li>
</ul>
</div><div class="column" style="width:50%;">
<figure>
<img data-src="figs/murillo/greatapes.png"
alt="Phylogeny of the great apes from Javier-Prado et al. (2013) Nature" />
<figcaption aria-hidden="true">Phylogeny of the great apes from
Javier-Prado et al. (2013) Nature</figcaption>
</figure>
<div class="caption flushright">
<p><a href="https://www.nature.com/articles/nature12228">Javier-Prado et
al. (2013)</a></p>
</div>
</div>
</div>
</section>
<section id="genetic-diversity-in-the-great-apes-chromosome-12"
class="slide level2">
<h2>Genetic diversity in the great apes: chromosome 12</h2>
<p><img data-src="figs/murillo/greatapes_pi.png" /></p>
</section>
<section id="genetic-divergence-between-the-great-apes-chromosome-12"
class="slide level2">
<h2>Genetic divergence between the great apes: chromosome 12</h2>
<p><img data-src="figs/murillo/greatapes_dxy.png" /></p>
</section>
<section id="goals" class="slide level2"
data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom" data-background-size="60%">
<h2 data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom" data-background-size="60%">Goals</h2>
<ul>
<li>How correlated are landscapes for closely related species?</li>
<li>How much is due to shared footprints of
<ul>
<li>history?</li>
<li>selection? what kind?</li>
<li>mutational processes?</li>
</ul></li>
</ul>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-positive.png"
class="invisible" style="height:80.0%" /></p>
</section>
<section id="slimulations" class="slide level2">
<h2>SLiMulations</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Chromosome-scale simulation (chr12)</li>
<li>selection on annotated exons (from Ensembl via <a
href="https://popsim-consortium.github.io/stdpopsim-docs/main/catalog.html#sec_catalog_HomSap_annotations">stdpopsim</a>)</li>
<li>deCode genetic map</li>
<li>branches simulated independently with SLiM and merged <a
href="https://tskit.dev/pyslim/docs/latest/vignette_parallel_phylo.html">tskit</a></li>
<li>runtime: hours to weeks</li>
<li>total simulations: 56</li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="figs/murillo/demog_tubes_greatapes.png" /></p>
</div>
</div>
</section>
<section id="back-to-the-data" class="slide level2">
<h2>Back to the data</h2>
<div class="centered">
<p><img data-src="figs/murillo/greatapes_pi.png" style="width:70.0%" />
<img data-src="figs/murillo/greatapes_dxy.png"
style="width:70.0%" /></p>
</div>
</section>
<section id="section-14" class="slide level2">
<h2></h2>
<p><img
data-src="figs/murillo/scatter-examples-pidxy_win-size_1000000_merged-mask_True_state_all_curr_all_prop-acc_0.4.png" /></p>
<p><img data-src="figs/murillo/trees-with-dt.png" /></p>
</section>
<section
id="the-data-empirical-correlations-for-chromosome-12-in-the-great-apes"
class="slide level2">
<h2>The data: empirical correlations for chromosome 12 in the great
apes</h2>
<p><img data-src="figs/murillo/cor-pidxy-dT_data.png" /></p>
</section>
<section id="no-correlation-under-neutrality" class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">No
correlation under neutrality</h2>
<div class="columns">
<div class="column" style="width:30%;">
<ul>
<li>correlations decay to zero quickly with split time</li>
</ul>
</div><div class="column" style="width:70%;">
<p><img data-src="figs/murillo/corr-pidxy-land-panel_neutral.png" /></p>
</div>
</div>
</section>
<section id="mutation-rate-variation-can-contribute"
class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">Mutation rate variation can contribute</h2>
<div class="columns">
<div class="column" style="width:30%;">
<ul>
<li>but correlations do not decay with time</li>
</ul>
</div><div class="column" style="width:70%;">
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-mutvar-sims_1.png"
class="centered" style="width:80.0%" /></p>
</div>
</div>
</section>
<section id="negative-selection-produces-weak-correlations"
class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">Negative selection produces weak
correlations</h2>
<p><img
data-src="figs/murillo/cor-pidxy-dT_selected-negative.png" /></p>
</section>
<section id="positive-selection-produces-strong-correlations"
class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">Positive selection produces strong
correlations</h2>
<ul>
<li>sometimes too strong</li>
</ul>
<div class="columns">
<div class="column" style="width:30%;">

</div><div class="column" style="width:70%;">
<p><span class="math display">\[ \vphantom{ d_{xy} = \pi_\text{anc}
\nearrow + \mu T_\text{MRCA} \searrow } \]</span></p>
</div>
</div>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-positive.png"
class="centered" /></p>
</section>
<section id="the-best-fit-both" class="slide level2"
data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right" data-background-size="30%">
<h2 data-background-image="figs/murillo/cor-pidxy-dT_data.png"
data-background-position="top 100px right"
data-background-size="30%">The best fit: both!</h2>
<ul>
<li>chosen from 37 distinct simulations</li>
<li>deleterious mutations: <span class="math inline">\(1.2 \times
10^{-8}\)</span></li>
<li>beneficial mutations: <span
class="math inline">\(10^{-12}\)</span></li>
</ul>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-both.png"
class="centered" style="width:60.0%" /></p>
</section>
<section id="conclusions" class="slide level2"
data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom" data-background-size="60%">
<h2 data-background-image="figs/murillo/landscapes_cover.png"
data-background-position="bottom"
data-background-size="60%">Conclusions</h2>
<ul>
<li>positive selection most necessary for a good fit</li>
<li>best guess: <span class="math inline">\(\approx 10\%\)</span> of
fixations on human lineage due to positive selection (1/250 years);</li>
<li><span class="math inline">\(\approx\)</span> 70% of mutations in
exons deleterious</li>
<li>fixation rate in exons reduced by half</li>
<li>GC-biased gene conversion causes “smile” at ends of chromosomes</li>
</ul>
<p><img data-src="figs/murillo/cor-pidxy-dT_selected-positive.png"
class="invisible" style="height:80.0%" /></p>
</section></section>
<section>
<section id="maps-of-population-density"
class="title-slide slide level1">
<h1>Maps of population density</h1>
<div class="caption floatright">
<p><img data-src="figs/mapNN/chris.jpeg" style="width:80.0%" /></p>
<p><a
href="https://scholar.google.com/citations?user=ZHHGVn8AAAAJ&amp;hl=en">Chris
C R Smith</a></p>
</div>
</section>
<section id="goal" class="slide level2">
<h2>Goal</h2>
<p><em>This</em> but with parameters <span
class="math inline">\(\mu\)</span>, <span
class="math inline">\(\theta\)</span> the <strong>maps</strong> of mean
density (<span class="math inline">\(K\)</span>) and dispersal distance
(<span class="math inline">\(\sigma\)</span>):</p>
<p><img data-src="figs/modeling_parameters_inverse_computer.png" /></p>
</section>
<section id="design" class="slide level2">
<h2>Design</h2>
<p><img data-src="figs/mapNN/architecture.png" /></p>
</section>
<section id="training" class="slide level2">
<h2>Training</h2>
<ul>
<li>randomly generated landscapes of <span
class="math inline">\(K\)</span> and <span
class="math inline">\(\sigma\)</span>,</li>
<li>passed to spatial SLiMulations</li>
<li>and used as the target</li>
<li>data: <span class="math inline">\(10^8\)</span>bp genomes</li>
<li>450 random sample pairs</li>
</ul>
</section>
<section id="results" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/feems_1a.png" /></p>
</section>
<section id="results-1" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/feems_1.png" /></p>
</section>
<section id="results-2" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/methods_comparison.png" /></p>
</section>
<section id="results-3" class="slide level2">
<h2>Results</h2>
<p><img data-src="figs/mapNN/methods_comparison.png" /></p>
<p>… stay tuned for a preprint!</p>
</section></section>
<section>
<section id="wrap-up" class="title-slide slide level1">
<h1>Wrap-up</h1>

</section>
<section id="other-applications" class="slide level2">
<h2>Other applications</h2>
<ul>
<li>history of <em>Nebria</em> beetles (Gilia Patterson, Sean Schoville,
Yi-Ming Weng)</li>
<li>inference of mean dispersal distance (Chris Smith; disperseNN)</li>
<li>and of maps of effective density and dispersal distance (Chris
Smith; mapNN)</li>
<li>inference of recombination rate maps (Jeff Adrion; ReLERNN)</li>
</ul>
<div class="fragment">
<ul>
<li>inference of maps of population density (Gilia Patterson)</li>
<li>and individual movement</li>
<li>identification of regions under selection</li>
</ul>
</div>
</section>
<section id="software-development-goals" class="slide level2">
<h2>Software development goals</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>open</li>
<li>welcoming and supportive</li>
<li>reproducible and well-tested</li>
<li>backwards compatible</li>
<li>well-documented</li>
<li>capacity building</li>
</ul>
<div class="centered">
<figure>
<img data-src="figs/popsim.png" style="width:35.0%" alt="popsim logo" />
<figcaption aria-hidden="true">popsim logo</figcaption>
</figure>
</div>
<p><a
href="https://popsim-consortium.github.io/stdpopsim-docs/stable/index.html">PopSim
Consortium</a></p>
</div><div class="column" style="width:50%;">
<div class="centered">
<figure>
<img data-src="figs/tskit_logo.png" style="width:60.0%"
alt="tskit logo" />
<figcaption aria-hidden="true">tskit logo</figcaption>
</figure>
<p><a href="https://tskit.dev">tskit.dev</a></p>
<figure>
<img data-src="figs/slim_logo.png" style="width:80.0%"
alt="SLiM logo" />
<figcaption aria-hidden="true">SLiM logo</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="thanks" class="slide level2">
<h2>Thanks!</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Andy Kern</li>
<li>Victoria Caudill</li>
<li>Murillo Rodrigues</li>
<li>Gilia Patterson</li>
<li>Chris Smith</li>
<li>Nate Pope</li>
<li>Jiseon Min</li>
<li>Clara Rehmann</li>
<li>Bruce Edelman</li>
<li>Anastasia Teterina</li>
<li>Matt Lukac <!--
- Saurabh Belsare
- Gabby Coffing
- Jeff Adrion
- CJ Battey
- Jared Galloway
--></li>
<li>the rest of <a href="https://kr-colab.github.io/people">the
Co-Lab</a></li>
</ul>
<p>Funding:</p>
<ul>
<li>NIH NIGMS</li>
<li>NSF DBI</li>
</ul>
</div><div class="column" style="width:50%;">
<div style="font-size: 85%; margin-top: -40px;">
<ul>
<li>Jerome Kelleher</li>
<li>Ben Haller</li>
<li>Ben Jeffery</li>
<li>Yan Wong</li>
<li>Elsie Chevy</li>
<li>Madeline Chase</li>
<li>Sean Stankowski</li>
<li>Matt Streisfeld <!--
- Georgia Tsambos
- Jaime Ashander
- Jared Galloway
- Gideon Bradburd
- Bill Cresko
- Alison Etheridge
- Evan McCartney-Melstad
- Brad Shaffer
--></li>
</ul>
</div>
<div class="floatright">
<p><img data-src="figs/KernRalph_5x5.png" style="width:20.0%" /> <img
data-src="figs/tskit_logo.png" style="width:30.0%" alt="tskit logo" />
<img data-src="figs/slim_logo.png" style="width:30.0%"
alt="SLiM logo" /> <img data-src="figs/colab.png"
style="width:70.0%" /></p>
</div>
</div>
</div>
</section>
<section id="section-15" class="slide level2"
data-background-image="figs/guillemots_thanks.png"
data-background-position="center" data-background-size="50%">
<h2 data-background-image="figs/guillemots_thanks.png"
data-background-position="center" data-background-size="50%"></h2>
<!-- Nebria -->
</section></section>
<section>
<section id="beetles-in-the-mountains" class="title-slide slide level1">
<h1>Beetles in the mountains</h1>

</section>
<section id="nebria-ingensriversi" class="slide level2">
<h2><em>Nebria ingens/riversi</em></h2>
<div class="columns">
<div class="column" style="width:40%;">
<ul>
<li>ground beetles</li>
<li>live on flowing snowmelt in the Sierra Nevada of California</li>
<li>cannot fly</li>
<li>live 1–2 years</li>
</ul>
</div><div class="column" style="width:60%;">
<div class="floatright">
<p><img data-src="figs/nebria/nebria_overview.png"
style="width:100.0%" /></p>
</div>
</div>
</div>
</section>
<section class="slide level2">

<div class="columns">
<div class="column" style="width:40%;">
<div class="centered">
<figure>
<img data-src="figs/nebria/LGM.png" style="width:100.0%"
alt="Nebria habitat in the LGM" />
<figcaption aria-hidden="true">Nebria habitat in the LGM</figcaption>
</figure>
</div>
</div><div class="column" style="width:60%;">
<p>Good habitat followed the glaciers uphill (SDM by Yi-Ming Weng).</p>
<div class="floatright">
<figure>
<img data-src="figs/nebria/sdm.png" style="width:100.0%" alt="SDM" />
<figcaption aria-hidden="true">SDM</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="goals-1" class="slide level2">
<h2>Goals</h2>
<ul>
<li>What was the spatial demographic history of <em>Nebria</em> since
the LGM?</li>
<li>Where did the ancestors of today’s beetles live in the past?</li>
<li>Were the ancestors of <em>N. ingens</em> and <em>N. riversi</em>
distinct at the LGM?</li>
</ul>
<div class="centered">
<figure>
<img data-src="figs/nebria/sdm.png" style="width:100.0%" alt="SDM" />
<figcaption aria-hidden="true">SDM</figcaption>
</figure>
</div>
</section>
<section id="data" class="slide level2">
<h2>Data</h2>
<div class="columns">
<div class="column" style="width:40%;">
<p>Collected by Yi-Ming Weng and Sean Schoville:</p>
<ul>
<li>384 beetles at 27 sites</li>
<li>low coverage WGS</li>
<li>some presence/absence data</li>
<li>expert knowledge about good locations</li>
<li>a species distribution model</li>
</ul>
</div><div class="column" style="width:60%;">
<div class="floatright">
<p><img data-src="figs/nebria/nebria_sampling.png"
style="width:100.0%" /></p>
</div>
</div>
</div>
<p><img data-src="figs/nebria/big_sdm.png" style="width:100.0%" /></p>
</section>
<section id="methods-overview" class="slide level2">
<h2>Methods overview</h2>
<div class="columns">
<div class="column" style="width:40%;">
<ol type="1">
<li>Develop an individual-based SLiMulation.</li>
</ol>
</div><div class="column" style="width:60%;">
<div class="floatright">
<p>SLiM screenshot</p>
</div>
</div>
</div>
</section>
<section id="methods-overview-1" class="slide level2">
<h2>Methods overview</h2>
<div class="columns">
<div class="column" style="width:40%;">
<ol type="1">
<li>Develop an individual-based SLiMulation.</li>
<li>Use presence-absence estimates to narrow the range of parameter
values based on 200 years of simulation.</li>
</ol>
</div><div class="column" style="width:60%;">
<div class="floatright">
<p>SLiM screenshot</p>
</div>
</div>
</div>
</section>
<section id="methods-overview-2" class="slide level2">
<h2>Methods overview</h2>
<div class="columns">
<div class="column" style="width:40%;">
<ol type="1">
<li>Develop an spatial, individual-based SLiMulation.</li>
<li>Do ABC on presence-absence estimates to narrow the range of
parameter values based on 200 years of simulation.</li>
<li>Run simulations since LGM, and do ABC on mean heterozygosity and
divergence.</li>
</ol>
</div><div class="column" style="width:60%;">
<div class="floatright">
<p>SLiM screenshot</p>
</div>
</div>
</div>
</section>
<section id="simulations" class="slide level2"
data-background-image="figs/nebria/sims1.png"
data-background-position="bottom 100px right"
data-background-size="100%">
<h2 data-background-image="figs/nebria/sims1.png"
data-background-position="bottom 100px right"
data-background-size="100%">Simulations</h2>
</section>
<section id="simulations-1" class="slide level2"
data-background-image="figs/nebria/sims2.png"
data-background-position="bottom 100px right"
data-background-size="100%">
<h2 data-background-image="figs/nebria/sims2.png"
data-background-position="bottom 100px right"
data-background-size="100%">Simulations</h2>
</section>
<section id="simulations-2" class="slide level2"
data-background-image="figs/nebria/sims3.png"
data-background-position="bottom 100px right"
data-background-size="100%">
<h2 data-background-image="figs/nebria/sims3.png"
data-background-position="bottom 100px right"
data-background-size="100%">Simulations</h2>
</section></section>
<section>
<section id="more-on-landscapes-of-diversity"
class="title-slide slide level1">
<h1>More on landscapes of diversity</h1>

</section>
<section class="slide level2">

<div class="centered">
<p><img data-src="figs/murillo/far_divergence_tree.png"
style="width:80.0%" /></p>
</div>
</section>
<section id="a-quick-primer" class="slide level2"
data-background-image="figs/murillo/far_divergence_tree.png"
data-background-position="top 200px right 50px"
data-background-size="30%">
<h2 data-background-image="figs/murillo/far_divergence_tree.png"
data-background-position="top 200px right 50px"
data-background-size="30%">A quick primer</h2>
<div class="columns">
<div class="column" style="width:70%;">
<ul>
<li class="fragment">Genetic diversity (<span
class="math inline">\(\pi\)</span>) and divergence (<span
class="math inline">\(d_{XY}\)</span>) are <em>time since common
ancestor</em> multiplied by <em>effective mutation rate</em>: <span
class="math display">\[ \pi, d_{XY} = T_\text{MRCA} \times
\mu_\text{effective} . \]</span></li>
<li class="fragment">Higher mutation rate <span
class="math inline">\(\Rightarrow\)</span> higher <span
class="math inline">\(\pi\)</span> and faster increase of <span
class="math inline">\(d_{XY}\)</span>.</li>
<li class="fragment">Positive selection <em>increases</em> fixation rate
of new mutations <span class="math inline">\(\Rightarrow\)</span> <span
class="math inline">\(d_{XY}\)</span> goes up faster,</li>
<li class="fragment">… but, it <em>decreases</em> <span
class="math inline">\(\pi\)</span> nearby.</li>
</ul>
</div>
</div>
</section>
<section class="slide level2">

<p><strong>linked selection:</strong> The indirect effects of selection
on genomic locations that are <em>linked</em> to the sites under
selection by a lack of recombination.</p>
<p>Selection tends to <em>decrease</em> diversity, over a distance
determined by recombination rate.</p>
<div class="centered">
<p><img data-src="figs/sweep_haps.png" class="fragment"
style="width:40.0%" /> <img data-src="figs/background_haps.png"
class="fragment" style="width:40.0%" /></p>
</div>
<!--
----------------------
![Langley et al 2012](figs/from_the_literature/langley-et-al-2012-chr3-pi-and-rho.png)
-->
</section>
<section id="diversity-correlates-with-recombination-rate"
class="slide level2">
<h2>Diversity correlates with recombination rate</h2>
<div class="columns">
<div class="column" style="width:80%;">
<figure>
<img
data-src="figs/from_the_literature/corbett-detig-divergence-recomb-all-species.png"
style="width:85.0%" alt="Corbett-Detig et al" />
<figcaption aria-hidden="true">Corbett-Detig et al</figcaption>
</figure>
</div><div class="column" style="width:20%;">
<p><em>Hudson 1994; Cutter &amp; Payseur 2013; Corbett-Detig et al
2015</em></p>
</div>
</div>
<!-- Divergence landscapes changing over time (from Murillo) -->
</section>
<section id="how-diversity-landscapes-change" class="slide level2">
<h2>How diversity landscapes change</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="figs/murillo/close_divergence_tree.png" /></p>
</div><div class="column" style="width:50%;">
<p><img
data-src="figs/murillo/two_spp_neutral_div_landscapes_0N.jpg" /></p>
</div>
</div>
</section>
<section id="how-diversity-landscapes-change-1" class="slide level2">
<h2>How diversity landscapes change</h2>
<div class="columns">
<div class="column" style="width:50%;">

</div><div class="column" style="width:50%;">
<p><img
data-src="figs/murillo/two_spp_neutral_div_landscapes.gif" /></p>
</div>
</div>
</section>
<section id="how-diversity-landscapes-change-2" class="slide level2">
<h2>How diversity landscapes change</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="figs/murillo/far_divergence_tree.png" /></p>
</div><div class="column" style="width:50%;">
<p><img
data-src="figs/murillo/two_spp_neutral_div_landscapes_4N.jpg" /></p>
</div>
</div>
</section>
<section id="how-can-landscapes-stay-correlated" class="slide level2">
<h2>How can landscapes stay correlated?</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>shared processes:
<ul>
<li>positive, negative selection</li>
<li>mutation rate variation</li>
<li>GC-biased gene conversion</li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="figs/murillo/two_spp_sel_div_landscapes_0N.jpg" /></p>
</div>
</div>
</section>
<section id="how-can-landscapes-stay-correlated-1" class="slide level2">
<h2>How can landscapes stay correlated?</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>shared processes:
<ul>
<li>positive, negative selection</li>
<li>mutation rate variation</li>
<li>GC-biased gene conversion</li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="figs/murillo/two_spp_sel_div_landscapes.gif" /></p>
</div>
</div>
</section>
<section id="how-can-landscapes-stay-correlated-2" class="slide level2">
<h2>How can landscapes stay correlated?</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>shared processes:
<ul>
<li>positive, negative selection</li>
<li>mutation rate variation</li>
<li>GC-biased gene conversion</li>
</ul></li>
</ul>
</div><div class="column" style="width:50%;">
<p><img data-src="figs/murillo/two_spp_sel_div_landscapes_4N.jpg" /></p>
</div>
</div>
<!-- End divergence landscapes changing over time -->
</section></section>
    </div>
  </div>

  <script src="./reveal.js/dist/reveal.js"></script>

  <!-- reveal.js plugins -->
  <script src="./reveal.js/plugin/notes/notes.js"></script>
  <script src="./reveal.js/plugin/search/search.js"></script>
  <script src="./reveal.js/plugin/zoom/zoom.js"></script>
  <script src="./reveal.js/plugin/math/math.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
        // Display controls in the bottom right corner
        controls: true,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: true,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'bottom-right',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: true,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: false,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: true,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'default',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: true,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'fade',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        math: {
          mathjax: '/usr/share/javascript/mathjax/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [
          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    </body>
</html>
